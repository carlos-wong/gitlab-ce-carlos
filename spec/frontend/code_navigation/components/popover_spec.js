import { shallowMount } from '@vue/test-utils';
import Popover from '~/code_navigation/components/popover.vue';

const MOCK_CODE_DATA = Object.freeze({
  hover: [
    {
      language: 'javascript',
      value: 'console.log',
    },
  ],
  definition_url: 'http://test.com',
});

const MOCK_DOCS_DATA = Object.freeze({
  hover: [
    {
      language: null,
      value: 'console.log',
    },
  ],
  definition_url: 'http://test.com',
});

let wrapper;

function factory(position, data) {
  wrapper = shallowMount(Popover, { propsData: { position, data } });
}

describe('Code navigation popover component', () => {
  afterEach(() => {
    wrapper.destroy();
  });

  it('renders popover', () => {
    factory({ x: 0, y: 0, height: 0 }, MOCK_CODE_DATA);

    expect(wrapper.element).toMatchSnapshot();
  });

  describe('code output', () => {
    it('renders code output', () => {
      factory({ x: 0, y: 0, height: 0 }, MOCK_CODE_DATA);

      expect(wrapper.find({ ref: 'code-output' }).exists()).toBe(true);
      expect(wrapper.find({ ref: 'doc-output' }).exists()).toBe(false);
    });
  });

  describe('documentation output', () => {
    it('renders code output', () => {
      factory({ x: 0, y: 0, height: 0 }, MOCK_DOCS_DATA);

      expect(wrapper.find({ ref: 'code-output' }).exists()).toBe(false);
      expect(wrapper.find({ ref: 'doc-output' }).exists()).toBe(true);
    });
  });
});
