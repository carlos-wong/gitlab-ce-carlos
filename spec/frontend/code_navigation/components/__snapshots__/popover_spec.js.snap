// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code navigation popover component renders popover 1`] = `
<div
  class="popover code-navigation-popover popover-font-size-normal gl-popover bs-popover-bottom show"
  style="left: 0px; top: 0px;"
>
  <div
    class="arrow"
    style="left: 0px;"
  />
   
  <div
    class="border-bottom"
  >
    <pre
      class="border-0 bg-transparent m-0 code highlight"
    >
      console.log
    </pre>
  </div>
   
  <div
    class="popover-body"
  >
    <gl-button-stub
      class="w-100"
      href="http://test.com"
      size="md"
      target="_blank"
      variant="default"
    >
      
      Go to definition
    
    </gl-button-stub>
  </div>
</div>
`;
